src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -806,6 +809,22 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		int op = id == SN_GetLower ? OP_XLOWER : OP_XUPPER;
 		return emit_simd_ins_for_sig (cfg, klass, op, 0, arg0_type, fsig, args);
 	}
+	case SN_Negate:
+	case SN_OnesComplement: {
+#ifdef TARGET_ARM64
+		int op = id == SN_Negate ? OP_ARM64_XNEG : OP_ARM64_MVN;
+		return emit_simd_ins_for_sig (cfg, klass, op, -1, arg0_type, fsig, args);
+#else
+		return NULL;
+#endif
+	}
+	case SN_Sqrt: {
+#ifdef TARGET_ARM64
+		return emit_simd_ins_for_sig (cfg, klass, OP_XOP_OVR_X_X, INTRINS_AARCH64_ADV_SIMD_FSQRT, arg0_type, fsig, args);
```

Comment:
Does `FSQRT` work for both floating point and integer types?

Reply:
No it doesn't cover, that is why there were failing tests. Changed it

Reply:
It seems that there isn't any Arm64 intrinsics for sqrt with integer types. @tannergooding Please correct me if I were wrong. Also, I am wondering if `uint32x4_t vrsqrteq_u32 (uint32x4_t a)` and `uint32x2_t vrsqrte_u32 (uint32x2_t a)` could possibly be leveraged here.

Reply:
> It seems that there isn't any Arm64 intrinsics for sqrt with integer types

Correct. The software implementation handles it by doing `integer type -> float type` then `sqrt` then `float type -> integer type`.

So you could accelerate here, if desirable, it would just be more work/logic to make that happen. RyuJIT isn't accelerating that pattern yet either.

> uint32x4_t vrsqrteq_u32 (uint32x4_t a) and uint32x2_t vrsqrte_u32 (uint32x2_t a)

Potentially under Mono's `fast-math` switch, but even then those instructions are "special" and I'd expect most users wouldn't want them (even under a fast-math opt-in). The general issue is that `vrqrte` is an "estimating" instructions and so while they are faster, they will also give inaccurate (often very inaccurate results). They are typically meant for rolling your own implementation that computes to the desired precision or for where you know the imprecision amount is acceptable.

Reply:
Make sense. To be aligned with CoreCLR should be sufficient for Mono atm. :)